### PR TITLE
feat(text-modal):Run text-modal tests

### PR DESCRIPTION
**Pull request by udhadpa**
Hello All,
Text-Modal tests are completed. Please review and let me know if you have any suggestions or corrections to make.

Thank You

@StommePoes @JasonLantz  @wyseguyonline @umahaea @mhomolak

### CODE REVIEW

**Comment by wyseguyonline** [comment]
LGTM

File: src/test/java/origamiV2Tests/TextModalTest.java [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+
+    }
+
+    @Test(testName = "Hit on 'Esc' Button test", groups = "desktop-regression")
+    private void hitEscKeyTest() throws Exception {
+        commonUtils.click(textModalPgObj.initiateBtn);
+        WebElement modal = driver.findElement(textModalPgObj.modal);
+        modal.sendKeys(Keys.ESCAPE);
+        actInitiateBtnVal = commonUtils.getAttributeValue(textModalPgObj.initiateBtn, "aria-expanded");
+        isModalExpanded = commonUtils.assertValue(actInitiateBtnVal, "false", "Click on 'X' Btn did not close the modal");
```
Comment (JasonLantz) [inline src/test/java/origamiV2Tests/TextModalTest.java]: this is an esc button test not an X button... return message should probably reflect that...

**Comment by JasonLantz** [comment]
looks good Payal....

File: src/main/java/utilities/CommonUtils.java [hunk 2]
```diff
@@ -252,10 +252,8 @@ public boolean assertCSSProperties(String propertyType, String expectedValue, Ob
         try {
             if (Arrays.asList(arr).contains(expectedValue)) {
                 isCSSPropertyPresent = true;
-
             } else {
                 isCSSPropertyPresent = false;
-
             }
```
Comment (StommePoes) [inline src/main/java/utilities/CommonUtils.java]: isCSSPropertyPreset = bool(Arrays.asList(arr).contains(expectedValue));?

File: src/test/java/origamiV2Tests/TextModalTest.java [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+        commonUtils.click(textModalPgObj.cancelBtn);
+        Assert.assertTrue(isModalExpanded);
+    }
+
+    // Safari -  tabbing needs manual settings which is not possible in selenium..
+    @Test(testName = "Tab on Initiating Button test", groups = "desktop-regression")
+    private void tabButtonTest() throws Exception {
+        if (browser.equals("safari")) {
+            throw new SkipException("Safari browser does not support tab functionality");
+        }
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/TextModalTest.java]: Didn't we eventually find a way around this problem? It was either for OSX a different keystroke set was used (optoin+tab?) or somehow the settings could be turned on (to "all controls").

File: src/main/java/origamiV2/fixtures/textModal/text-modal.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="../../../elements/css/elements.css"/>
+    <link rel="shortcut icon" href="http://example.com/myicon.ico"/>
```
Comment (umahaea) [inline src/main/java/origamiV2/fixtures/textModal/text-modal.html]: is this icon 'shortcut icon' needed?

File: src/main/java/utilities/BaseClass.java [hunk 5]
```diff
@@ -103,42 +103,46 @@ else if (mobile.equals("on")) {
...
-                    appium = new AndroidDriver(new URL(URL), caps);
+                try {
+                    if (appiumDriver.equalsIgnoreCase("iOS")) {
+                        appium = new IOSDriver(new URL(URL), caps);
+                    } else if (appiumDriver.equalsIgnoreCase("android")) {
+                        appium = new AndroidDriver(new URL(URL), caps);
+                    }
+                    includePageObjects();
+                } catch (Exception e) {
+                    System.out.println(e);
```
Comment (umahaea) [inline src/main/java/utilities/BaseClass.java]: what kind of exception do you hit here? shouldn't it be e.getmessage?

File: src/test/java/origamiV2Tests/TextModalTest.java [hunk 6]
```diff
@@ -0,0 +1,548 @@
...
+        if (!browser.equals("chrome")) {
+            throw new SkipException("browser console logs apis are not yet implemented for this browserdriver'");
+        }
+        Thread.sleep(1000);
+        readInitialConfig(textModalJSPath);
+        changeConfig(textModalJSPath, defaultConfig, Config);
+        Thread.sleep(1000);
+        commonUtils.getUrl(textModalUrl);
+        browserLogs = commonUtils.browserLogs().toString();
+        System.out.println(browserLogs);
```
Comment (umahaea) [inline src/test/java/origamiV2Tests/TextModalTest.java]: can we remove this Sysout?

**Comment by umahaea** [comment]
rest looks good to me.

**Comment by udhadpa** [comment]
Hello All,

Fixed code as per your feedback

@JasonLantz  @StommePoes @umahaea

**Comment by StommePoes** [comment]
looks good.

**Comment by udhadpa** [comment]
All checks have been passed. Merging the branch into master

**Pull request merged**